File: android/src/main/kotlin/com/example/design_engine/layer2_core/dark/EOnSecondaryContainerDark.kt
package com.example.design_engine.layer2_core.dark
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnSecondaryContainerDark {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyd28
            AppTheme.STANDARD -> R.color.cstd28
            AppTheme.BLUE -> R.color.cbd28
            AppTheme.GREEN -> R.color.cgd28
            AppTheme.RED -> R.color.crd28
            AppTheme.NOTHING -> R.color.cnod28
        }
    }
}

File: lib/layer2_core/dark/e_on_secondary_container_dark.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/dark/system.dart';
import '../../layer1_raw/dark/standard.dart';
import '../../layer1_raw/dark/blue.dart';
import '../../layer1_raw/dark/green.dart';
import '../../layer1_raw/dark/red.dart';
import '../../layer1_raw/dark/nothing.dart';

class EOnSecondaryContainerDark {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyd28;
      case 'standard': return cstd28;
      case 'blue': return cbd28;
      case 'green': return cgd28;
      case 'red': return crd28;
      case 'nothing': return cnod28;
      default: return cstd28;
    }
  }
}
